apm-agent-plugins/apm-ratpack-plugin/src/main/java/co/elastic/apm/agent/ratpack/AbstractDefaultExecutionInstrumentation.java

```diff
@@ -0,0 +1,121 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.ratpack;
+
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import ratpack.exec.Execution;
+
+import java.util.function.Consumer;
+
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+public abstract class AbstractDefaultExecutionInstrumentation extends AbstractRatpackInstrumentation {
+
+    private final ElementMatcher<? super MethodDescription> methodMatcher;
+
+    public AbstractDefaultExecutionInstrumentation(ElementMatcher<? super MethodDescription> methodMatcher) {
+        this.methodMatcher = methodMatcher;
+    }
+
+    @Override
+    public final ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return named("ratpack.exec.internal.DefaultExecution");
+    }
+
+    @Override
+    public final ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return methodMatcher;
+    }
+
+    public static class RatpackActivateTransactionInstrumentation extends AbstractDefaultExecutionInstrumentation {
+        public RatpackActivateTransactionInstrumentation() {
+            super(named("bindToThread")
+                .and(isPublic())
+                .and(takesArguments(0))
+                .and(returns(TypeDescription.VOID)));
+        }
+
+        @SuppressWarnings("unused")
+        @Advice.OnMethodExit(suppress = Throwable.class)
+        public static void onAfterBind(@Advice.This Execution execution) {
+            activate(execution);
+        }
+
+        @VisibleForAdvice
+        public static void activate(Execution execution) {
+            execution.maybeGet(Transaction.class).ifPresent(new Consumer<Transaction>() {
+                @Override
+                public void accept(Transaction transaction) {
+
+                    if (tracer == null) {
+                        return;
+                    }
+
+                    if (tracer.currentTransaction() == null) {
+                        transaction.activateInScope();
```

Comment:
as you are not using the scope, just call `activate()` which does not allocate a scope object